test/test_corpus.cc

```diff
@@ -495,6 +495,129 @@ TEST_P(ExpectationTest, PerPhaseTest) { // NOLINT
+            got["ast"].append(file.tree->toString(*gs)).append("\n");
+        }
+
+        expectation = test.expectations.find("ast-raw");
+        if (expectation != test.expectations.end()) {
+            got["ast-raw"].append(file.tree->showRaw(*gs)).append("\n");
+        }
+
+        // DSL pass
+        file = testSerialize(*gs, ast::ParsedFile{dsl::DSL::run(ctx, move(file.tree)), file.file});
+
+        expectation = test.expectations.find("dsl-tree");
+        if (expectation != test.expectations.end()) {
+            got["dsl-tree"].append(file.tree->toString(*gs)).append("\n");
+        }
+
+        expectation = test.expectations.find("dsl-tree-raw");
+        if (expectation != test.expectations.end()) {
+            got["dsl-tree-raw"].append(file.tree->showRaw(*gs)).append("\n");
+        }
+
+        // local vars
+        file = testSerialize(*gs, local_vars::LocalVars::run(ctx, move(file)));
+
+        // namer
+        {
+            core::UnfreezeSymbolTable symbolTableAccess(*gs);
+            vector<ast::ParsedFile> vTmp;
+            vTmp.emplace_back(move(file));
+            vTmp = namer::Namer::run(ctx, move(vTmp));
+            file = testSerialize(*gs, move(vTmp[0]));
+        }
+
+        expectation = test.expectations.find("name-tree");
+        if (expectation != test.expectations.end()) {
+            got["name-tree"].append(file.tree->toString(*gs)).append("\n");
+        }
+
+        expectation = test.expectations.find("name-tree-raw");
+        if (expectation != test.expectations.end()) {
+            got["name-tree-raw"].append(file.tree->showRaw(*gs));
+        }
+
+        newTrees.emplace_back(move(file));
+    }
+
+    // resolver
+    trees = resolver::Resolver::runTreePasses(ctx, move(newTrees));
+    auto newErrors = errorQueue->drainAllErrors();
+    errors.insert(errors.end(), make_move_iterator(newErrors.begin()), make_move_iterator(newErrors.end()));
+
+    for (auto &resolvedTree : newTrees) {
+        expectation = test.expectations.find("resolve-tree");
+        if (expectation != test.expectations.end()) {
+            got["resolve-tree"].append(resolvedTree.tree->toString(*gs)).append("\n");
+        }
+
+        expectation = test.expectations.find("resolve-tree-raw");
+        if (expectation != test.expectations.end()) {
+            got["resolve-tree-raw"].append(resolvedTree.tree->showRaw(*gs)).append("\n");
+        }
+    }
+
+    for (auto &gotPhase : got) {
+        auto expectation = test.expectations.find(gotPhase.first);
+        ASSERT_TRUE(expectation != test.expectations.end()) << "missing expectation for " << gotPhase.first;
+        ASSERT_TRUE(expectation->second.size() == 1)
+            << "found unexpected multiple expectations of type " << gotPhase.first;
+
+        auto checker = test.folder + expectation->second.begin()->second;
+        auto expect = FileOps::read(checker.c_str());
+        EXPECT_EQ(expect, gotPhase.second) << "Mismatch on: " << checker;
```

Comment:
Is it clear at all from the error messages that the failure is happening in the "stress incremental resolver" part of the tests? These errors look like they might be indistinguishable from regular PosTest failures.

Reply:
One possibility is using `SCOPED_TRACE`:

https://github.com/google/googletest/blob/master/googletest/docs/advanced.md#adding-traces-to-assertions

...while another is just changing the error message ("Mismatch on" => "[incremental-resolver] Mismatch on")

I did the latter approach for fast path assertions.